include/RAJA/exec-cuda/reduce_cuda.hxx

```diff
@@ -654,43 +663,53 @@ class ReduceSum<cuda_reduce<BLOCK_SIZE>, T> {
   __device__ ReduceSum<cuda_reduce<BLOCK_SIZE>, T> operator+=(T val) const {
     __shared__ T sd[BLOCK_SIZE];
 
-    if (blockDim.x * blockIdx.x + threadIdx.x == 0) {
-      m_max_grid_size[0] = RAJA_MAX(gridDim.x, m_max_grid_size[0]);
+    int blockId = blockIdx.x 
+       + blockIdx.y * gridDim.x 
+       + gridDim.x * gridDim.y * blockIdx.z; 
+
+
+    int threadId = threadIdx.x + 
+               blockDim.x * threadIdx.y + 
+               (blockDim.x * blockDim.y) * threadIdx.z;
+
+
+    if (blockId  + threadId == 0) {
+      m_max_grid_size[0] = RAJA_MAX(gridDim.x*gridDim.y*gridDim.z, m_max_grid_size[0]);
```

Comment:
Can we add an "after-the-fact" assertion in the destructor or casting method to verify that m_max_grid_size is actually smaller than the amount of memory we allocated for the sum reduction?  This would be better than the current state of never knowing if an internal buffer has overflowed.